aicsimageio/transforms.py

```diff
@@ -93,71 +33,180 @@ def reshape_data(
+    # Check for parameter conflicts
+    for dim in Dimensions.DefaultOrderList:
+        # return_dims='TCZYX' and fixed dimensions 'C=1'
+        # Dimension is in kwargs
+        # Dimension is an integer
+        # Dimension is in return dimensions
+        if isinstance(kwargs.get(dim), int) and dim in return_dims:
+            raise ConflictingArgumentsError(
+                f"When selecting a single dimension index, the specified dimension can "
+                f"not be provided in return_dims. "
+                f"return_dims={return_dims}, dimension {dim} = {kwargs.get(dim)}"
+            )
 
-    # Check for conflicts like return_dims='TCZYX' and fixed channels 'C=1'
-    for dim in return_dims:
-        if kwargs.get(dim) is not None:
+        # return_dims='CZYX' and iterable dimensions 'T=range(10)'
+        # Dimension is in kwargs
+        # Dimension is an iterable
+        # Dimension is not in return dimensions
+        if (
+            isinstance(kwargs.get(dim), (list, tuple, range, slice))
+            and dim not in return_dims
+        ):
             raise ConflictingArgumentsError(
-                f"Argument return_dims={return_dims} and "
-                f"argument {dim}={kwargs.get(dim)} conflict. "
-                f"Check usage."
+                f"When selecting a multiple dimension indicies, the specified "
+                f"dimension must be provided in return_dims. "
+                f"return_dims={return_dims}, dimension {dim} = {kwargs.get(dim)}"
             )
 
-    # Add each dimension not included in original data
+    # Process each dimension available
     new_dims = given_dims
-    excluded_dims = "".join(set(return_dims) - set(given_dims))
-    for dim in excluded_dims:
-        # Dask doesn't have an explicit expand dims so
-        # we simply reshape with an extra dim
-        data = operator.reshape(data, (1, *data.shape))
-        new_dims = dim + new_dims  # add the missing Dimension to the front
-
-    # If given dims contains a Dimension not in return dims and its depth is 1 remove it
-    # If it's larger than 1 give a warning and
-    # suggest interfacing with the Reader object
-    extra_dims = "".join(set(given_dims) - set(return_dims))
-    for dim in extra_dims:
-        index = new_dims.find(dim)
-        if data.shape[index] > 1:
-            index_depth = kwargs.get(dim)
-            if index_depth is None:
+    dim_specs = []
+    for dim in given_dims:
+        # Store index of the dim as it is in given data
+        dim_index = given_dims.index(dim)
+
+        # Handle dim in return dims which means that it is
+        # an iterable or None selection
+        if dim in return_dims:
+            # Specific iterable requested
+            if dim in kwargs:
+                # Actual dim specification
+                # The specification provided for this dimension in the kwargs
+                dim_spec = kwargs.get(dim)
+                display_dim_spec = dim_spec
```

Comment:
Possible rewording (same on L159 and L169)
```suggestion
                display_dim_spec = dim_spec = kwargs.get(dim)
```

Reply:
I personally don't like these lines. I understand what they are doing but its the verbose vs like first glance understanding thing for me.

(And I always worry about typing an extra `=` to turn the right side into `==`)

Reply:
All good, just a thought :) It wasn't immediately clear to me what `display_dim_spec` was used for and I was concerned that an operation had accidentally been missed on that assignment line. This makes it clear that the multiple assignment was intentional, but it's totally fine to keep it as-is.